File: 백준/Gold/2212. 센서/센서.py
import sys
input = sys.stdin.readline
N = int(input())
K = int(input())
ls = list(map(int, input().split()))

if N <= K:
    ans = 0
else:
    ls.sort()
    distances = []
    for i in range(N - 1):
        tmp = ls[i + 1] - ls[i]
        distances.append(tmp)
    distances.sort(reverse=True)
    ans = sum(distances[K-1:])
print(ans)

File: 백준/Gold/2212. 센서/센서.js
const fs = require("fs");
const path = require("path");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");
let N = parseInt(input[0]);
let K = parseInt(input[1]);
let ls = input[2].split(" ").map(Number);
let ans = 0;

if (N > K){
  ls.sort((a, b) => a - b);
  let distances = [];
  for (let i = 0; i < N-1; i++){
    distances.push(ls[i+1] - ls[i]);
  }
  distances.sort((a, b) => b - a);
  let tmp = distances.slice(K-1);
  ans = tmp.reduce((acc, cur) => acc + cur, 0);
}
console.log(ans);
